source/adios2/core/Engine.h

```diff
@@ -242,9 +250,10 @@ class Engine
      * must use Read(variable) instead intentionally
      */
     template <class T>
-    void Read(Variable<T> &variable, const T *values)
+    void Read(Variable<T> &variable, T *values)
     {
-        Read(variable, values);
+        ScheduleRead(variable, values);
+        PerformReads(PerformReadMode::BLOCKINGREAD);
```

Comment:
This is not clear yet. Yes we separate them out, so ScheduleRead() and PerformRead() are official. Question if we will have Read() as schedule+perform(in blocking mode) or rather we also eliminate Write() and have ScheduleWrite() instead. We are sill undecided on this.